index.js

```diff
@@ -21,4 +21,160 @@ window.addEventListener('resize', () => {
+    source: 'https://github.com/YoutLone/Portfolio',
+  },
+];
+
+// Get the modal element
+const modal = document.querySelector('.modal');
+
+// Function to update the modal content
+function updateModal(project) {
+  // Update the modal content with project details
+  const modalContent = modal.querySelector('.modal-content');
+  modalContent.querySelector('.main-project').textContent = project.projectName;
+  modalContent.querySelector('.main-description').textContent = project.description;
+  modalContent.querySelector('.modal-text').textContent = project.modaltext;
+
+  const tagsContainer = modalContent.querySelector('.main-tags');
+  tagsContainer.innerHTML = '';
+  project.tags.forEach((tag) => {
+    const tagElement = document.createElement('li');
+    tagElement.className = 'main-tag';
+    tagElement.textContent = tag;
+    tagsContainer.appendChild(tagElement);
+  });
+
+  modalContent.querySelector('.image').src = project.image;
+
+  const seeLiveBtn = modalContent.querySelector('.see-btn:nth-child(1)');
+  seeLiveBtn.href = project.live;
+
+  const seeSourceBtn = modalContent.querySelector('.see-btn:nth-child(2)');
+  seeSourceBtn.href = project.source;
+}
+
+// Attach event listeners to each images
+const projectElements = document.querySelectorAll('.click-img');
+projectElements.forEach((projectElement) => {
+  projectElement.addEventListener('click', (e) => {
+    // Show the modal
+    modal.style.display = 'flex';
+    modal.style.position = 'fixed';
+    const content = projects[e.currentTarget.dataset.position];
+    // Update the modal with the corresponding project details
+    updateModal(content);
+    window.scrollTo(0, 0);
+  });
+});
+
+// Function to close the modal
+function closeModal() {
+  modal.style.display = 'none';
+}
+
+// Close the modal when the close button is clicked
+const closeBtn = modal.querySelector('.close');
+closeBtn.addEventListener('click', closeModal);
+
+// Close the modal when the user clicks outside of it
+window.addEventListener('click', (event) => {
+  if (event.target === modal) {
+    closeModal();
+  }
+});
+
+// Validate-Contact_Form
+
+const email = document.querySelector('#email');
+const formBtn = document.querySelector('.form-btn');
+const errorMessage = document.querySelector('.error-message');
+const sentMessage = document.querySelector('.sent-message');
+const contactForm = document.querySelector('.contact-form');
+
+function isItUppercase(value) {
+  if (value.match(/^[a-z@.0-9-_]*$/)) {
+    return true;
+  }
+  return false;
+}
+
+formBtn.addEventListener('click', (e) => {
+  if (email.value !== '' && isItUppercase(email.value)) {
+    sentMessage.textContent = ' Message is successfully send!';
+    email.style.border = '1px solid green';
+    contactForm.submit();
+    contactForm.reset();
+  } else {
+    e.preventDefault();
+    errorMessage.textContent = 'Your email should be in lowerCase';
+    email.style.border = '2px solid red';
+  }
+});
```

Comment:
- [ ] Seems your forms are behaving abnormally. Two error messages are displayed at the same time with a success message, creating a bad user experience. This happen when I submit the form after entering characters into the email field completely different from the email characters. I suggest you fix the issues are you can see in the screenshot below:
![Screenshot from 2023-05-19 10-49-52](https://github.com/YoutLone/Portfolio/assets/63087491/8d9f4a52-7ebf-4639-9c70-e96b581ffcbf)